crates/stackable-webhook/src/servers/conversion.rs

```diff
@@ -19,141 +66,246 @@ where
-    /// [`crate::servers::ConversionReview`].
+    /// 1. The CRD
+    /// 2. A conversion function to convert between CRD versions. Typically you would use the
+    ///   the auto-generated `try_convert` function on CRD spec definition structs for this.
+    ///
+    /// The [`ConversionWebhookServer`] takes care of reconciling the CRDs into the Kubernetes
+    /// cluster and takes care of adding itself as conversion webhook. This includes TLS
+    /// certificates and CA bundles.
     ///
     /// # Example
     ///
-    /// ```
+    /// ```no_run
     /// use stackable_webhook::{
     ///     servers::{ConversionReview, ConversionWebhookServer},
     ///     Options
     /// };
+    /// use stackable_operator::cli::OperatorEnvironmentOpts;
+    /// use stackable_operator::kube::Client;
+    /// use stackable_operator::crd::s3::{S3Connection, S3ConnectionVersion};
+    ///
+    /// # async fn test() {
+    /// let crds_and_handlers = [
+    ///     (
+    ///         S3Connection::merged_crd(S3ConnectionVersion::V1Alpha1)
+    ///             .expect("failed to merge S3Connection CRD"),
+    ///         S3Connection::try_convert as fn(_) -> _,
+    ///     ),
+    /// ];
+    ///
+    /// const OPERATOR_NAME: &str = "PRODUCT_OPERATOR";
+    /// let client = Client::try_default().await.expect("failed to create Kubernetes client");
+    /// // Normally you would get this from the CLI arguments in
+    /// // `ProductOperatorRun::operator_environment`
+    /// let operator_environment = OperatorEnvironmentOpts {
+    ///     operator_namespace: "stackable-operator".to_string(),
+    ///     operator_service_name: "product-operator".to_string(),
+    /// };
     ///
     /// // Construct the conversion webhook server
-    /// let server = ConversionWebhookServer::new(handler, Options::default());
+    /// let conversion_webhook = ConversionWebhookServer::new(
+    ///     crds_and_handlers,
+    ///     stackable_webhook::Options::default(),
+    ///     client,
+    ///     OPERATOR_NAME,
+    ///     operator_environment,
+    /// )
+    /// .await
+    /// .expect("failed to create ConversionWebhookServer");
     ///
-    /// // Define the handler function
-    /// fn handler(req: ConversionReview) -> ConversionReview {
-    ///    // In here we can do the CRD conversion
-    ///    req
-    /// }
+    /// conversion_webhook.run().await.expect("failed to run ConversionWebhookServer");
+    /// # }
     /// ```
-    #[instrument(name = "create_conversion_webhhok_server", skip(handler))]
-    pub fn new<H>(handler: H, options: Options) -> Self
+    #[instrument(
+        name = "create_conversion_webhook_server",
+        skip(crds_and_handlers, client)
+    )]
+    pub async fn new<H>(
+        crds_and_handlers: impl IntoIterator<Item = (CustomResourceDefinition, H)>,
+        options: Options,
+        client: Client,
+        field_manager: impl Into<String> + Debug,
+        operator_environment: OperatorEnvironmentOpts,
```

Comment:
note: A good place for this might be the `Options` for `ConversionWebhookServer`.